contracts/pallet-ibc/src/ics20/memo.rs

```diff
@@ -186,16 +184,16 @@ impl<T: Config + Send + Sync, S: Module + Clone + Default + PartialEq + Eq + Deb
 
 impl<T: Config + Send + Sync, S: Module + Clone + Default + PartialEq + Eq + Debug> Memo<T, S> {
 	fn process_memo(packet: &mut Packet) -> Result<(), Error> {
-		let packet_data: PacketData =
-			serde_json::from_slice(packet.data.as_slice()).map_err(|e| {
-				Error::implementation_specific(format!("Failed to decode packet data {:?}", e))
-			})?;
-		let receiver = <T as Config>::AccountIdConversion::try_from(packet_data.receiver.clone())
-			.map_err(|_| {
-				Error::implementation_specific(format!("Failed to parse receiver account"))
-			})?
-			.into_account();
-		<T as Config>::HandleMemo::execute_memo(&packet_data, receiver).map_err(|e| {
+		// let packet_data: PacketData =
```

Comment:
do we remove these?

Reply:
yes. we removed and moved the logic about 
`let receiver = <T as Config>::AccountIdConversion::try_from(packet_data.receiver.clone())`
into execute_memo
to NOT call `serde_json::from_slice(packet.data.as_slice())` twice
here and inside of execute memo function